bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Path.java

```
@@ -125,11 +120,10 @@ public Path (Device device, Path path, float flatness) {
 	if (path == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
 	if (path.isDisposed()) SWT.error(SWT.ERROR_INVALID_ARGUMENT);
 	flatness = Math.max(0, flatness);
-	long handle = Gdip.GraphicsPath_Clone(path.getHandle(path.initialZoom));
-	if (flatness != 0) Gdip.GraphicsPath_Flatten(handle, 0, flatness);
-	if (handle == 0) SWT.error(SWT.ERROR_NO_HANDLES);
-	initialZoom = path.initialZoom;
-	zoomToHandle.put(initialZoom, new PathHandle(handle, initialZoom));
+	operations.addAll(path.operations);
+	if (flatness != 0) {
+		operations.add(new FlattenOperation(flatness));
```

This comment depends on the other general comment on error handling: In order to not hide potential failures, should we do `storeAndApply...` here? E.g., could passing a negative flatness yield an error that may pop up on access instead of instantiation with this change?

---

Negative flatness is not possible because of line 122.
He is definitely no handle available, so storeAndApply would do nothing except storing it. But out of consistency it still makes sense to do that here, because if there would be handles, it should probably be applied here.

---

I adapted these places to use storeAndApply instead